[19.0] [MIG] document_url: Migration to 19.0

BhaveshHeliconia opened the pull request.

pedrobaeza committed:
[ADD] document_url: Module to attach URLs as documents

sysadminmatmoz committed:
Slovene translations added

Commit:
[MIG] from 8.0 to 9.0 document_url (#89)

[MIG] document_url

* bump version to 9.0.1.0.0 and make it installable
* Update JS structure
* Add css in order to be compliant with odoo style
* Add a README.rst and OCA conventions
* Improve selector
* add fr.po for fr translation
* open link with target='_blank'

pedrobaeza committed:
[FIX] document_url: Fix documentation

oca-transbot committed:
OCA Transbot updated translations from Transifex

yajo committed:
[FIX][document_url] Consistent styling.

Without this patch, when you hover on the menu item, its background color does not change, and its cursor is a text one.

With this patch, the menu item styling matches those available in all surrounding menus.

ernestotejeda committed:
[MIG] document_url: Migration to 11.0

ernestotejeda committed:
[MIG] document_url: Refactoring code

Replace char © by word Copyright
Change format in usage description

oca-travis committed:
[UPD] Update document_url.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

MiquelRForgeFlow committed:
[FIX] pylint/javascript-lint

ernestotejeda committed:
[MIG] document_url: Migration to 12.0

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update document_url.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: knowledge-12.0/knowledge-12.0-document_url
Translate-URL: https://translation.odoo-community.org/projects/knowledge-12-0/knowledge-12-0-document_url/

OCA-git-bot committed:
[UPD] README.rst

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 100.0% (14 of 14 strings)

Translation: knowledge-12.0/knowledge-12.0-document_url
Translate-URL: https://translation.odoo-community.org/projects/knowledge-12-0/knowledge-12-0-document_url/hr/

Commit:
[IMP] document_url: black, isort

Commit:
[MIG] document_url: Migration to 13.0

oca-travis committed:
[UPD] Update document_url.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: knowledge-13.0/knowledge-13.0-document_url
Translate-URL: https://translation.odoo-community.org/projects/knowledge-13-0/knowledge-13-0-document_url/

sergio-teruel committed:
[FIX] document_url: Fix module dependencies.

OCA-git-bot committed:
document_url 13.0.1.0.1

DavidBForgeFlow committed:
[MIG] document_url: Migration to 14.0

[MIG] document_url: Migration to 14.0

victoralmau committed:
[MIG] document_url: Migration to 14.0
TT30263

victoralmau committed:
[IMP] document_url: Visual and functional improvements in UI.
Define the add attachment and add URL buttons on the same line.
Allow downloading of url type attachments by clicking on the icon or the name.
Define the name of the url type attachments as links (allows to open it in a new tab).
Hide the download button for url attachments.
Force to set mimetype to "application/link" for url attachments.
Displays a link icon for url attachments.
TT30263

JoelZilli committed:
[IMP] document_url: pre-commit execution

matiasperalta1 committed:
[MIG] document_url: Migration to 15.0

Commit:
[UPD] Update document_url.pot

OCA-git-bot committed:
[UPD] README.rst

FernandoRomera committed:
[16.0][MIG] document_url: Migration to 16.0

Commit:
[UPD] Update document_url.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: knowledge-16.0/knowledge-16.0-document_url
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_url/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (14 of 14 strings)

Translation: knowledge-16.0/knowledge-16.0-document_url
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_url/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (14 of 14 strings)

Translation: knowledge-16.0/knowledge-16.0-document_url
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_url/it/

pedrobaeza committed:
[FIX] document_url: Change mimetype only if type=url

Other modules like web_pwa_oca uses url field for storing an indexed
path for later being searched in the controller, so let's restrict the
mimetype change if both url and type=url is set.

OCA-git-bot committed:
document_url 16.0.1.0.1

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (14 of 14 strings)

Translation: knowledge-16.0/knowledge-16.0-document_url
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_url/it/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 92.8% (13 of 14 strings)

Translation: knowledge-16.0/knowledge-16.0-document_url
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_url/sl/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (14 of 14 strings)

Translation: knowledge-16.0/knowledge-16.0-document_url
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_url/it/

luis-ron committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (14 of 14 strings)

Translation: knowledge-16.0/knowledge-16.0-document_url
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_url/es/

etobella committed:
[FIX] document_url: Migration was not done properly

Commit:
[UPD] Update document_url.pot

OCA-git-bot committed:
document_url 16.0.1.0.2

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: knowledge-16.0/knowledge-16.0-document_url
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_url/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (17 of 17 strings)

Translation: knowledge-16.0/knowledge-16.0-document_url
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_url/it/

angelinaanaki committed:
[16.0][FIX] document_url: Add URL button and paperclip button

OCA-git-bot committed:
document_url 16.0.1.0.3

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (17 of 17 strings)

Translation: knowledge-16.0/knowledge-16.0-document_url
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_url/es/

OCA-git-bot committed:
[UPD] README.rst

DemchukM committed:
[16.0][FIX]document_url: fix Not working properly when "web_responsive" is installed

OCA-git-bot committed:
[BOT] post-merge updates

peterromao committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (17 of 17 strings)

Translation: knowledge-16.0/knowledge-16.0-document_url
Translate-URL: https://translation.odoo-community.org/projects/knowledge-16-0/knowledge-16-0-document_url/pt/

lef-adhoc committed:
[IMP] document_url: pre-commit auto fixes

lef-adhoc committed:
[MIG] document_url: Migration to 17.0

Commit:
[UPD] Update document_url.pot

OCA-git-bot committed:
[BOT] post-merge updates

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (17 of 17 strings)

Translation: knowledge-17.0/knowledge-17.0-document_url
Translate-URL: https://translation.odoo-community.org/projects/knowledge-17-0/knowledge-17-0-document_url/sv/

esteveblanch committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (17 of 17 strings)

Translation: knowledge-17.0/knowledge-17.0-document_url
Translate-URL: https://translation.odoo-community.org/projects/knowledge-17-0/knowledge-17-0-document_url/ca/

maq-adhoc committed:
[FIX] document_url: Fix Download attachment component

Fix download atachment widget.

Document_url when activated causes error when opening attachment link on the chatter window
This commit (https://github.com/odoo/odoo/commit/366676cafdce00d55823c6daf41452b0c2373e4d) move download buttons into a sub-component.
I modified the module behavior to use canDownload() method instead of adding a t-if in the view and prevent the chatter component from breaking when the message has attachments.

OCA-git-bot committed:
[BOT] post-merge updates

luisDIXMIT committed:
[IMP] document_page_partner: pre-commit auto fixes

BhaveshHeliconia committed:
[IMP] document_url: pre-commit auto fixes

BhaveshHeliconia committed:
[MIG] document_url: Migration to 18.0

Commit:
[UPD] Update document_url.pot

OCA-git-bot committed:
[BOT] post-merge updates

BhaveshHeliconia committed:
[MIG] document_url: Migration to 19.0

pedrobaeza commented:
/ocabot migration document_url

pedrobaeza commented on document_url/tests/test_document_url.py:
```diff
@@ -0,0 +1,46 @@
+    def setUp(self):
+        super().setUp()
+        wizard_add_url = self.env["ir.attachment.add_url"]
+        self.user = self.env["res.users"].create(
```
Use `from odoo.tests import new_test_user`

pedrobaeza reviewed (changes requested):
Tested on runboat. Are you testing your migrations in your end before pushing them to a PR? It's not working correctly for clicking on the attachment to go to the URL:

<img width="719" height="219" alt="Image" src="https://github.com/user-attachments/assets/da00bdbc-15b1-4ff7-9d18-78e90f6d6e72" />

There shouldn't preview box as well.

And finally, please squash administrative commits into the main one.

BhaveshHeliconia commented:
@pedrobaeza I did test the migration on my side, and at that time it was working for me. After reviewing it again, I noticed the issue occurs after refreshing the page once the link is added—most likely I missed that scenario during my initial testing.

I’ve now fixed the issue and also improved the UI for a better user experience. Apologies for the inconvenience.

pedrobaeza commented:
OK, now it's working, but it seems you have copied/pasted the whole CSS. Can you reduce it with inheritance?

ypapouin committed:
[FIX] document_url: Easier template inheriting

- In a first step, instead of replacing a node content with its verbatim copy, we uses the xpath "replace" command with "$0" magic keyword to move all the <t> content into a <div>.
- In a second step, the "Add URL" button is added after the existing "Attach files"button.

pedrobaeza commented:
Please apply #581 (before migration commit).

marcelsavegnago reviewed (approved):
LGTM

pedrobaeza commented:
/ocabot merge nobump
